### PR TITLE
Slack

### PR DESCRIPTION
**Pull request by ledamint**

### CODE REVIEW

File: src/utils/formatHTMLElement.js [hunk 1]
```diff
@@ -18,45 +18,72 @@ export function formatHTMLElement(element) {
...
     if (React.isValidElement(element)) {
-        const text = formatHTMLElement(element.props.children);
+        const text = formatHTMLElement(element.props.children, parseMode);
 
         switch (element.type) {
             case React.Fragment: {
                 return text;
             }
             case 'b': {
+                if (parseMode === 'markdown') {
```
Comment (oTkPoBeHuE) [inline src/utils/formatHTMLElement.js]:
``` tsx
function formatMarkupLanguageElement(element, markupLanguage= 'HTML'){
   if(markupLanguage=== 'HTML'){
      return formatHTMLElement(element);
   }

  if(markupLanguage=== 'Markdown'){
    return formatMarkdown(element);
  }
}

function formatHTMLElement(element){
  switch (element.type) {

  }
}

function formatMarkdownElement(element){
  switch (element.type) {

  }
}
```

File: src/components/Image.js [hunk 2]
```diff
@@ -24,20 +30,22 @@ export function Image(props) {
...
 
-    let parseMode = parseModeProp;
+    let parseMode = parseModeProp ?? parseModeContext;
     let title = titleElement;
-    if (typeof titleElement !== 'string' && typeof titleElement !== 'number') {
-        parseMode = 'HTML';
-        title = formatHTMLElement(titleElement);
+
+    if (typeof children !== 'string' && typeof children !== 'number') {
+        parseMode = parseMode ?? 'HTML';
```
Comment (oTkPoBeHuE) [inline src/components/Image.js]: double protection :) 
parseMode ?? 'HTML' and Inside formatHTMLElement

File: src/components/ButtonGroup.js [hunk 3]
```diff
@@ -14,15 +14,21 @@ export function ButtonGroup(props) {
...
     const buttons = formatButtonElement(children);
 
-    let parseMode = parseModeProp;
+    let parseMode = parseModeProp ?? parseModeContext;
     let title = titleElement;
-    if (typeof titleElement !== 'string' && typeof titleElement !== 'number') {
-        parseMode = 'HTML';
-        title = formatHTMLElement(titleElement);
+
+    if (typeof children !== 'string' && typeof children !== 'number') {
```
Comment (oTkPoBeHuE) [inline src/components/ButtonGroup.js]: Duplicate code. See Image.js

File: src/utils/formatHTMLElement.js (hunk 1 above)
Comment (ledamint) [reply src/utils/formatHTMLElement.js]: Good point, fixed

File: src/components/Image.js (hunk 2 above)
Comment (ledamint) [reply src/components/Image.js]: remove the default from format function

File: src/components/ButtonGroup.js (hunk 3 above)
Comment (ledamint) [reply src/components/ButtonGroup.js]: good point, fixed

**Pull request merged**